src/NewChain/ChainVerifier.ts

```diff
@@ -0,0 +1,464 @@
+    this.validateExpectedAndDeployedAddress(
+      this.auxiliaryWeb3,
+      organization,
+      this.contractAddresses.auxiliary.ostCoGatewayOrganizationAddress,
+      'OSTPrime: Invalid organization address!!!'
+    );
+
+    const coGateway = await ostPrimeInstance.methods.coGateway().call();
+    this.validateExpectedAndDeployedAddress(
+      this.auxiliaryWeb3,
+      coGateway,
+      this.contractAddresses.auxiliary.ostEIP20CogatewayAddress,
+      'OSTPrime: Invalid coGateway address!!!'
+    );
+
+    Logger.info('Successfully completed OSTPrime contract verification!!!');
+  }
+
+  /**
+   * Check if deployed bin is valid or not
+   * @param web3 Web3 endpoint.
+   * @param contractName Name of the contract to verify.
+   * @param contractAddress Contract address to verify.
+   * @param errMsg Error message to log and throw.
+   */
+  private async validateBIN(web3, contractName, contractAddress, errMsg) {
+    const deployedBin = await web3.eth.getCode(contractAddress);
+    if (this.abiBinProvider.getBIN(contractName).toLowerCase().indexOf(
+        deployedBin.toLowerCase().slice(2),
+      ) === -1
+    ) {
+      this.logAndThrowError(
+        errMsg,
+        { contractName: contractName, contractAddress: contractAddress}
+        );
+    }
+  }
+
+  /**
+   * Check if deployed bin is valid or not
+   * @param web3 Web3 endpoint.
+   * @param contractName Name of the contract to verify.
+   * @param contractAddress Contract address to verify.
+   * @param messageBusLinkInfo Linked bin info of message bus contract.
+   * @param gatewayLibLinkInfo Linked bin info of gateway lib contract.
+   * @param errMsg Error message to log and throw.
+   */
+  private async validateLinkedBin(
+    web3,
+    contractName,
+    contractAddress,
+    messageBusLinkInfo,
+    gatewayLibLinkInfo,
+    errMsg) {
+    const gatewayLinkedBin = this.abiBinProvider.getLinkedBIN(
+      contractName,
+      messageBusLinkInfo,
+      gatewayLibLinkInfo,
+    );
+    const deployedBin = await web3.eth.getCode(contractAddress);
+    if (gatewayLinkedBin.toLowerCase().indexOf(deployedBin.slice(2).toLowerCase()) === -1) {
+     this.logAndThrowError(errMsg);
+    }
+  }
+
+  /**
+   * Check if deployed bin is valid or not
+   * @param web3 Web3 endpoint.
+   * @param contractName Name of the contract to verify.
+   * @param contractAddress Contract address to verify.
+   * @param errMsg Error message to show.
+   */
+  private validateExpectedAndDeployedAddress(
+    web3,
+    deployedAddress,
+    mosaicConfigAddress,
+    errMsg) {
+
+    if (web3.utils.toChecksumAddress(deployedAddress)
+      !== web3.utils.toChecksumAddress(mosaicConfigAddress)) {
+      this.logAndThrowError(errMsg, {expected: mosaicConfigAddress, deployed: deployedAddress});
+    }
+  }
+
+  /**
+   * Log error and throw message.
+   *
+   * @param errMsg Error message.
+   * @param options Optional data to log.
+   */
+  private logAndThrowError(errMsg, options = {}) {
```

Comment:
We can just have 
`Logger.error(errMsg, options);`
`throw new Error(errMsg);`

Instead of calling this function.